test-harness/src/test/groovy/com/netflix/conductor/test/resiliency/QueueResiliencySpec.groovy

```diff
@@ -0,0 +1,605 @@
+        0 * queueDAO._
+        notThrown(Exception)
+        pollResult == null
+    }
+
+    def "Verify updateTask with COMPLETE status succeeds when QueueDAO is unavailable"() {
+        when: "Start a simple workflow"
+        def workflowInstanceId = workflowResource.startWorkflow(new StartWorkflowRequest()
+                .withName(SIMPLE_TWO_TASK_WORKFLOW)
+                .withVersion(1))
+
+        then: "Verify workflow is started"
+        with(workflowResource.getExecutionStatus(workflowInstanceId, true)) {
+            status == Workflow.WorkflowStatus.RUNNING
+            tasks.size() == 1
+            tasks[0].taskType == 'integration_task_1'
+            tasks[0].status == Task.Status.SCHEDULED
+        }
+
+        when: "The first task 'integration_task_1' is polled"
+        def task = taskResource.poll("integration_task_1", "test", null)
+
+        then: "Verify task is returned successfully"
+        task
+        task.status == Task.Status.IN_PROGRESS
+        task.taskType == 'integration_task_1'
+
+        when: "the above task is updated, while QueueDAO is unavailable"
+        def taskResult = new TaskResult(task)
+        taskResult.setStatus(TaskResult.Status.COMPLETED)
+        def result = taskResource.updateTask(taskResult)
+
+        then: "updateTask returns successfully without any exceptions"
+        1 * queueDAO.remove(*_) >> { throw new IllegalStateException("Queue remove failed from Spy") }
+        result == task.getTaskId()
+        notThrown(Exception)
+    }
+
+    def "Verify updateTask with IN_PROGRESS state fails when QueueDAO is unavailable"() {
+        when: "Start a simple workflow"
+        def workflowInstanceId = workflowResource.startWorkflow(new StartWorkflowRequest()
+                .withName(SIMPLE_TWO_TASK_WORKFLOW)
+                .withVersion(1))
+
+        then: "Verify workflow is started"
+        with(workflowResource.getExecutionStatus(workflowInstanceId, true)) {
+            status == Workflow.WorkflowStatus.RUNNING
+            tasks.size() == 1
+            tasks[0].taskType == 'integration_task_1'
+            tasks[0].status == Task.Status.SCHEDULED
+        }
+
+        when: "The first task 'integration_task_1' is polled"
+        def task = taskResource.poll("integration_task_1", "test", null)
+
+        then: "Verify task is returned successfully"
+        task
+        task.status == Task.Status.IN_PROGRESS
+        task.taskType == 'integration_task_1'
+
+        when: "the above task is updated, while QueueDAO is unavailable"
+        def taskResult = new TaskResult(task)
+        taskResult.setStatus(TaskResult.Status.IN_PROGRESS)
+        taskResult.setCallbackAfterSeconds(120)
+        def result = taskResource.updateTask(taskResult)
+
+        then: "updateTask returns successfully without any exceptions"
+        2 * queueDAO.postpone(*_) >> { throw new IllegalStateException("Queue postpone failed from Spy") }
```

Comment:
Why is postpone invoked twice? Given that we retry twice, shouldn't this be 3?

Reply:
The RetryUtil we're using here retries up to the provided number, which should be greater than >=1.